Marlin/src/lcd/extui/lib/mks_ui/draw_preHeat.cpp

```diff
@@ -216,12 +216,12 @@ void disp_desire_temp() {
 
   if (uiCfg.curTempType == 0) {
     strcat(public_buf_l, uiCfg.extruderIndex < 1 ? preheat_menu.ext1 : preheat_menu.ext2);
-    sprintf(buf, preheat_menu.value_state, (int)thermalManager.degHotend(uiCfg.extruderIndex), (int)thermalManager.degTargetHotend(uiCfg.extruderIndex));
+    sprintf(buf, preheat_menu.value_state, thermalManager.degHotend(uiCfg.extruderIndex), thermalManager.degTargetHotend(uiCfg.extruderIndex));
```

Comment:
you need cast to int here if the code uses `%d`... in fact, is safe to keep the cast! I think we need/should keep the cast to int.